pylops/utils/_internal.py

```diff
@@ -1,6 +1,49 @@
+from functools import wraps
+
 import numpy as np
 
 
+def reshape_flatten(forward=True):
+    """Decorator for the common reshape/flatten pattern used in many operators.
+
+    Example
+    =======
+    A ``_matvec`` (forward) function can be simplified from
+
+    .. code-block:: python
+
+        def _matvec(self, x):
+            x = np.reshape(x, self.dims)
+            y = do_things_to_reshaped(y)
+            y = y.ravel()
+            return y
+
+    to
+
+    .. code-block:: python
+        @reshape_flatten()
+        def _matvec(self, x):
+            y = do_things_to_reshaped(y)
+            return y
+
+    Similarly, an ``_rmatvec`` (adjoint) function can be simplified with by calling
+    ``reshape_flatten`` with ``forward=False``.
+
+    """
+    inp_dims = "dims" if forward else "dimsd"
+
+    def decorator(func):
+        @wraps(func)
+        def wrapper(self, x):
+            x = x.reshape(getattr(self, inp_dims))
+            y = func(self, x)
+            return y.ravel()
+
+        return wrapper
+
+    return decorator
```

Comment:
this looks very good, probably we should add this routine to the documentation in the PyLops utilities (it is true it is only for developers but it is something a bit more `exposed` than things like `_value_or_list_like_to_array` and it may start to be present in most operators.

What do you think?